stock_release_channel/models/stock_picking.py

```diff
@@ -81,9 +83,16 @@ def _delay_assign_release_channel(self):
     def assign_release_channel(self):
         messages = ""
         for pick in self:
-            result = self.env["stock.release.channel"].assign_release_channel(pick)
+            log_stream = io.StringIO()
+            result = self.env["stock.release.channel"].assign_release_channel(
+                pick.with_context(job_log_stream=log_stream)
```

Comment:
```suggestion
                pick.with_context(assign_release_channel_log_stream=log_stream)
```
Inside the method there is no job dependency, we return a string (that is handled by the job running the method but we don't care).
As context keys are globals, better to put a meaningful name maybe?